pyomo/core/plugins/transform/add_slack_vars.py

```diff
@@ -5,11 +5,40 @@
 
 from pyomo.common.modeling import unique_component_name
 from pyomo.core.plugins.transform.hierarchy import NonIsomorphicTransformation
+from pyomo.common.config import ConfigBlock, ConfigValue
+from pyomo.core.base.component import ComponentUID
+from pyomo.core.base import Constraint, _ConstraintData
+from pyomo.common.deprecation import deprecation_warning
 
-from random import randint
-
-import logging
-logger = logging.getLogger('pyomo.core')
+def target_list(x):
+    deprecation_msg = ("In future releases ComponentUID targets will no longer "
+                       "be supported. Specify targets as a Constraint or list "
+                       "of Constraints.")
+    if isinstance(x, ComponentUID):
+        deprecation_warning(deprecation_msg)
```

Comment:
We should probably only emit the deprecation message once for every call to the transformation.  Maybe something like:
```python
if deprecation_msg:
    deprecation_warning(deprecation_msg)
    deprecation_msg = None
```